source/frontend/src/components/common/GenericModal.tsx

```diff
@@ -215,12 +217,19 @@ export const GenericModal = (props: Omit<BsModalProps, 'onHide'> & ModalProps) =
                 title="cancel-modal"
                 variant={cancelButtonVariant ?? 'secondary'}
                 onClick={close}
+                test-id="cancel-modal-button"
```

Comment:
we already have the title property to get the buttons in tests. not opposed to adding test ids but I believe the right syntax is **data-testid="..."** 

See here: https://testing-library.com/docs/queries/bytestid/